test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs

```diff
@@ -4189,108 +4188,247 @@ public static TheoryData<CreateTokenTheoryData> IncludeSecurityTokenOnFailureTes
                 },
             };
         }
-    }
 
-    public class CreateTokenTheoryData : TheoryDataBase
-    {
-        public CreateTokenTheoryData()
+        [Theory, MemberData(nameof(ValidateAuthenticationTagLengthTheoryData))]
+        public void ValidateTokenAsync_ModifiedAuthNTag(CreateTokenTheoryData theoryData)
         {
+            // arrange
+            AppContext.SetSwitch(AuthenticatedEncryptionProvider._skipValidationOfAuthenticationTagLength, theoryData.EnableAppContextSwitch);
+            var payload = new JObject()
```

Comment:
Consider putting the payload, jsonWebTokenHandler, and signingCredentials into the theory data.

Reply:
Appreciate the idea.. All the theory data cases share the same payload, handler, and signing creds and there would be more code duplication to pull them out into the theory datas. 😵 Please let me know if there's another reason, otherwise I will keep as is 🤔

Reply:
If you define the vars in the method like you already do with signingCredentials512 and then assign the variables to the CreateTokenTheoryData members "Payload", "JsonWebTokenHandler", and "SigningCredentials" while there would be 3 more lines per theory data, it would make the test cleaner. Currently 13 of the 30 lines (43%) in your test are just defining these variables and aren't adding value to reader comprehension.

Ultimately your choice, imho since theory data is meant to hold the setup while the test is meant to hold the logic it seems to make more conceptual sense to put the vars in the theory data since it already has members for them.